src/backend/Functions/Analyze.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Fx.Portability;
+using Microsoft.Fx.Portability.ObjectModel;
+using Newtonsoft.Json;
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Functions
+{
+    public static class Analyze
+    {
+        [FunctionName("analyze")]
+        public static async Task<HttpResponseMessage> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequestMessage req,
+            TraceWriter log)
+        {
+            var analyzeRequest = await DeserializeRequest(req.Content);
+            if (analyzeRequest == null)
+            {
+                log.Error("invalid request");
+                return req.CreateResponse(HttpStatusCode.BadRequest);
+            }
+
+            var submissionId = Guid.NewGuid().ToString();
+            log.Info($"created submission id {submissionId}");
+
+            var response = req.CreateResponse(HttpStatusCode.OK);
+            response.Content = new StringContent(submissionId);
+
+            return response;
+        }
+
+        public static async Task<AnalyzeRequest> DeserializeRequest(HttpContent content)
+        {
+            try
+            {
+                var body = await DecompressContent(content);
```

Comment:
It's better to deserialize from stream rather than a string because we used to have a bunch of exceptions for strings being too long when the analysis request was huge.
I believe there is an extension [`DataExtensions.Deserialize<T>`](https://github.com/Microsoft/dotnet-apiport/blob/dev/src/lib/Microsoft.Fx.Portability/DataExtensions.cs#L21) that may do what you want.